src/OrbitQt/AnnotatingSourceCodeHandlerTest.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include <QCoreApplication>
+#include <QPushButton>
+#include <QTest>
+#include <QTimer>
+#include <filesystem>
+
+#include "AnnotatingSourceCodeHandler.h"
+#include "CodeReport/AnnotateDisassembly.h"
+#include "CodeReport/AnnotatingLine.h"
+#include "CodeReport/DisassemblyReport.h"
+#include "CodeViewer/OwningDialog.h"
+#include "OrbitBase/ExecutablePath.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ReadFileToString.h"
+#include "SourcePathsMapping/Mapping.h"
+#include "SourcePathsMapping/MappingManager.h"
+#include "SyntaxHighlighter/X86Assembly.h"
+#include "capture_data.pb.h"
+
+using namespace std::string_view_literals;
+const std::string_view kMainFunctionInstructions =
+    "\x50\xbf\x04\x20\x40\x00\xe8\xe5\xfe\xff\xff\x31\xc0\x59\xc3\x90"sv;
+
+constexpr const char* kOrgName = "The Orbit Authors";
+
+TEST(AnnotatingSourceCodeHandler, SmokeTest) {
+  QCoreApplication::setOrganizationName(kOrgName);
+  QCoreApplication::setApplicationName("AnnotatingSourceCodeHandler.SmokeTest");
+
+  const std::filesystem::path file_path =
+      orbit_base::GetExecutableDir() / "testdata" / "line_info_test_binary";
+
+  {
+    orbit_source_paths_mapping::MappingManager manager{};
+    manager.SetMappings({});
+    manager.AppendMapping(orbit_source_paths_mapping::Mapping{
+        std::filesystem::path{"."} / "..", orbit_base::GetExecutableDir() / "testdata"});
+  }
+
+  auto program = orbit_object_utils::CreateElfFile(file_path);
+  ASSERT_TRUE(program.has_value()) << program.error().message();
```

Comment:
I don't understand this. You are forwarding that error_message if the program doesn't have any value?

Reply:
gtest will show the given error message if the assertion fails. That's a pattern we have in almost every test. Here is an example in the official documentation: https://google.github.io/googletest/primer.html#assertions

Reply:
Yes, I saw them a lot but I wasn't 100% sure. Thanks for that!